sdk/src/main/scala/io/horizen/account/state/ForgerStakeStorageV3.scala

```diff
@@ -0,0 +1,404 @@
+    w.putBytes(s.stakedAmount.toByteArray)
+  }
+
+  override def parse(r: Reader): StakeCheckpoint = {
+    val fromEpochNumber = r.getInt()
+    val stakeAmountLength = r.getInt()
+    val stakeAmount = new BigIntegerUInt256(r.getBytes(stakeAmountLength)).getBigInt
+    StakeCheckpoint(fromEpochNumber, stakeAmount)
+  }
+}
+
+object ForgerMap {
+  val forgerList: StateDbArray = new StateDbArray(ACCOUNT, "Forgers".getBytes("UTF-8"))
+
+  def existsForger(view: BaseAccountStateView, forgerKey: Array[Byte]): Boolean = {
+    val forgerData = view.getAccountStorage(ACCOUNT, forgerKey)
+    // getting a not existing key from state DB using RAW strategy
+    // gives an array of 32 bytes filled with 0, while using CHUNK strategy
+    // gives an empty array instead
+    !forgerData.sameElements(NULL_HEX_STRING_32)
+  }
+
+  def addForger(view: BaseAccountStateView,
+                forgerKey: Array[Byte],
+                blockSignProposition: PublicKey25519Proposition,
+                vrfPublicKey: VrfPublicKey,
+                rewardShare: Int,
+                rewardAddress: Address,
+               ): Unit = {
+    forgerList.append(view, forgerKey)
+
+    val forgerStakeData = ForgerInfoV3(
+      ForgerPublicKeys(blockSignProposition, vrfPublicKey), rewardShare, new AddressProposition(rewardAddress))
+
+    // store the forger stake data
+    view.updateAccountStorageBytes(ACCOUNT, forgerKey,
+      ForgerInfoV3Serializer.toBytes(forgerStakeData))
+  }
+
+  def updateForger(view: BaseAccountStateView,
+                forgerKey: Array[Byte],
+                blockSignProposition: PublicKey25519Proposition,
+                vrfPublicKey: VrfPublicKey,
+                rewardShare: Int,
+                rewardAddress: Address,
+               ): Unit = {
+
+    val forgerStakeData = ForgerInfoV3(
+      ForgerPublicKeys(blockSignProposition, vrfPublicKey), rewardShare, new AddressProposition(rewardAddress))
+
+    // store the forger stake data
+    view.updateAccountStorageBytes(ACCOUNT, forgerKey,
+      ForgerInfoV3Serializer.toBytes(forgerStakeData))
+  }
+
+
+  def getSize(view: BaseAccountStateView): Int = forgerList.getSize(view)
+
+  def getKeys(view: BaseAccountStateView): Seq[Array[Byte]] = {
+    val listSize = getSize(view)
+    (0 until listSize).map(index => {
+     forgerList.getValue(view, index)
+    })
+  }
+
+  def getForger(view: BaseAccountStateView, forgerKey: Array[Byte]): Option[ForgerInfoV3] = {
+    val forgerData = view.getAccountStorage(ACCOUNT, forgerKey)
+    if (!forgerData.sameElements(NULL_HEX_STRING_32))
+      Some(ForgerInfoV3Serializer.parseBytes(view.getAccountStorageBytes(ACCOUNT, forgerKey)))
+    else
+      None
+  }
+
+  def apply(view: BaseAccountStateView, forgerKey: Array[Byte]): ForgerInfoV3 = {
+    val forgerData = view.getAccountStorage(ACCOUNT, forgerKey)
+    ForgerInfoV3Serializer.parseBytes(view.getAccountStorageBytes(ACCOUNT, forgerKey))
+  }
```

Comment:
forgerData is not used here, I believe we should check the existance as in getForger()

Reply:
You're right, I eliminated forgerData. 
As for checking for the existence, this method was created specifically for avoiding the overhead of comparing the result of view.getAccountStorageBytes(ACCOUNT, forgerKey) with NULL_HEX_STRING_32, as in getForger, because it is used in places where I'm expecting the forger exists, as when I'm iterating through the list of all forgers, as in `getAllForgerStakes`. At least for now, if there is a key inside the forger list, it must exist the corresponding forgerData in the storage.